diff -ur Trac-1.0.1-py2.7.egg.orig/trac/ticket/api.py Trac-1.0.1-py2.7.egg/trac/ticket/api.py
--- Trac-1.0.1-py2.7.egg.orig/trac/ticket/api.py	2014-02-10 13:26:26.618699400 +0100
+++ Trac-1.0.1-py2.7.egg/trac/ticket/api.py	2014-02-10 13:27:10.332199600 +0100
@@ -29,6 +29,19 @@
 from trac.util.translation import _, N_, gettext
 from trac.wiki import IWikiSyntaxProvider, WikiParser
 
+class ITicketChangelogFilter(Interface):
+    """Extension point interface for components that filter
+    changelog entries when they are being retrieved for display."""
+
+    def changelog_filter(req, changelog):
+        """
+        Called when a list of changelog entries is being retrieved
+        for display.
+        MUST return a list of changelog entries
+
+        Each changelog entry has the following fields:
+          date, author, field, old, new, permanent
+        """
 
 class ITicketActionController(Interface):
     """Extension point interface for components willing to participate
diff -ur Trac-1.0.1-py2.7.egg.orig/trac/ticket/web_ui.py Trac-1.0.1-py2.7.egg/trac/ticket/web_ui.py
--- Trac-1.0.1-py2.7.egg.orig/trac/ticket/web_ui.py	2014-02-10 13:26:26.840712100 +0100
+++ Trac-1.0.1-py2.7.egg/trac/ticket/web_ui.py	2014-02-10 13:27:10.442205900 +0100
@@ -34,7 +34,7 @@
     get_resource_shortname
 )
 from trac.search import ISearchSource, search_to_sql, shorten_result
-from trac.ticket.api import TicketSystem, ITicketManipulator
+from trac.ticket.api import TicketSystem, ITicketManipulator, ITicketChangelogFilter
 from trac.ticket.model import Milestone, Ticket, group_milestones
 from trac.ticket.notification import TicketNotifyEmail
 from trac.timeline.api import ITimelineEventProvider
@@ -70,6 +70,7 @@
     implements(IContentConverter, INavigationContributor, IRequestHandler,
                ISearchSource, ITemplateProvider, ITimelineEventProvider)
 
+    changelog_filters = ExtensionPoint(ITicketChangelogFilter)
     ticket_manipulators = ExtensionPoint(ITicketManipulator)
 
     timeline_details = BoolOption('timeline', 'ticket_show_details', 'false',
@@ -1685,7 +1686,8 @@
         in a `dict` object.
         """
         attachment_realm = ticket.resource.child('attachment')
-        for group in self.grouped_changelog_entries(ticket, when=when):
+        # NOTE -- ADDED req as parameter in the call
+        for group in self.grouped_changelog_entries(req, ticket, when=when):
             t = ticket.resource(version=group.get('cnum', None))
             if 'TICKET_VIEW' in req.perm(t):
                 self._render_property_changes(req, ticket, group['fields'], t)
@@ -1767,7 +1769,8 @@
                                 old=tag.em(old), new=tag.em(new))
         return rendered
 
-    def grouped_changelog_entries(self, ticket, db=None, when=None):
+    # NOTICE THE 'req' parameter
+    def grouped_changelog_entries(self, req, ticket, db=None, when=None):
         """Iterate on changelog entries, consolidating related changes
         in a `dict` object.
 
@@ -1776,6 +1779,11 @@
         """
         field_labels = TicketSystem(self.env).get_ticket_field_labels()
         changelog = ticket.get_changelog(when=when)
+        # BEGIN FILTERING OF CHANGELOG ENTRIES
+        for clogfilter in self.changelog_filters:
+            changelog = clogfilter.changelog_filter(req, changelog)
+        # END FILTERING OF CHANGELOG ENTRIES
+
         autonum = 0 # used for "root" numbers
         last_uid = current = None
         for date, author, field, old, new, permanent in changelog:
